### PR TITLE
3P Media: Properly map media3p urls to the "sizes" format expected by the app

### LINKED ISSUE #3065: Properly map media3p imageUrls to "full", "large" and "web_stories_thumbnail" values

### PR DESCRIPTION
**Pull request by diegovar**
## Summary

Properly map media3p urls to the "sizes" format expected by the app
## To-do

--

## User-facing changes

None (behind a flag).

## Testing Instructions

To be tested once the media3p integration is complete.

---

Fixes #3065

### CODE REVIEW

File: assets/src/edit-story/app/media/media3p/api/test/useMedia3pApi.js [hunk 1]
```diff
@@ -77,27 +112,69 @@ describe('useMedia3pApi', () => {
...
           sizes: {
+            web_stories_thumbnail: {
+              file: 'media/unsplash:1234',
+              source_url:
+                'https://images.unsplash.com/photo-1515516089376-88db1e26e9c0?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEzNzE5M30&fm=jpg&w=340&h=227&fit=max',
+              mime_type: 'image/jpeg',
+              width: 340,
+              height: 227,
+            },
+            '1060_707': {
```
Comment (swissspidy) [inline assets/src/edit-story/app/media/media3p/api/test/useMedia3pApi.js:132]: 1060_707, 2121_1414, 3182_2121 sounds a bit random. What are they used for specifically?

File: assets/src/edit-story/app/media/media3p/api/test/useMedia3pApi.js [hunk 2]
```diff
@@ -77,27 +112,69 @@ describe('useMedia3pApi', () => {
...
+          mimeType: 'image/jpeg',
           poster: null,
           posterId: null,
           sizes: {
+            web_stories_thumbnail: {
+              file: 'media/unsplash:1234',
+              source_url:
+                'https://images.unsplash.com/photo-1515516089376-88db1e26e9c0?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEzNzE5M30&fm=jpg&w=340&h=227&fit=max',
+              mime_type: 'image/jpeg',
+              width: 340,
```
Comment (swissspidy) [inline assets/src/edit-story/app/media/media3p/api/test/useMedia3pApi.js:129]: `web_stories_thumbnail` is defined as 150px wide, and not 340

File: assets/src/edit-story/app/media/media3p/api/test/useMedia3pApi.js (hunk 1 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/api/test/useMedia3pApi.js:132]: The API returns a minimum of 340 (which is the width of the media library), and then increments of 20%, 40%, 60%, 80% and 100% of the image's full size. The objective is to allow us to use srcsets properly, by using the images that make sense depending on where they will be displayed and the device's dpr. Today, the web_stories_thumbnail width of 150px results in blurry pictures when the mediia gallery is displayed in rows (eg: when the image width spans the entire gallery), and also for devices with dpr > 1 (eg: a Macbook Pro). To be honest, I'd like to not use the keys of the "sizes" object at all, and simply choose the URL that is best suited for each situation. So the getThumbnailUrl.js file would change to instead return the appropriate url for the given width (in the case of the media gallery, the width of it). WDYT?

File: assets/src/edit-story/app/media/media3p/api/test/useMedia3pApi.js (hunk 2 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/api/test/useMedia3pApi.js:129]: Commented above on our plan for image widths.

File: assets/src/edit-story/app/media/utils/getResourceFromMedia3p.js [hunk 3]
```diff
@@ -66,39 +66,38 @@ import createResource from './createResource';
...
+      throw new Error(
+        'Invalid number of urls for asset. Need at least 3: ' + m
+      );
     }
-    // Reverse sort and get the largest width to determine the full asset.
-    return m.imageUrls.sort((el1, el2) => el2.width - el1.width)[0];
+
+    const sizesFromBiggest = m.imageUrls
+      .sort((el1, el2) => el2.width - el1.width)
+      .map((u) => {
```
Comment (cvolzke4) [inline assets/src/edit-story/app/media/utils/getResourceFromMedia3p.js]: [nit]: This can be:

.map((u) => ({
  file: m.name,
  .....
}));

**Review by cvolzke4 — APPROVED** [review]

**Pull request merged**